shared/libraries/config_protocol/src/config_mirrored_ext_sig_impl.cpp

```diff
@@ -0,0 +1,131 @@
+#include <config_protocol/config_mirrored_ext_sig_impl.h>
+
+namespace daq::config_protocol
+{
+
+ConfigMirroredExternalSignalImpl::ConfigMirroredExternalSignalImpl(const ContextPtr& ctx,
+                                                                   const ComponentPtr& parent,
+                                                                   const StringPtr& remoteGlobalId)
+    : MirroredSignalBase(ctx, parent, createLocalId(remoteGlobalId), nullptr)
+    , remoteGlobalId(remoteGlobalId)
+{
+}
+
+StringPtr ConfigMirroredExternalSignalImpl::onGetRemoteId() const
+{
+    return remoteGlobalId;
+}
+
+StringPtr ConfigMirroredExternalSignalImpl::createLocalId(const StringPtr& remoteGlobalId)
+{
+    static constexpr char delimeter = '*';
+    std::string localId = remoteGlobalId;
+
+    const char slash = '/';
+    std::replace(localId.begin(), localId.end(), slash, delimeter);
+
+    return String(localId);
+}
+
+Bool ConfigMirroredExternalSignalImpl::onTriggerEvent(const EventPacketPtr& eventPacket)
+{
+    if (!eventPacket.assigned())
+        return False;
+
+    if (eventPacket.getEventId() == event_packet_id::DATA_DESCRIPTOR_CHANGED)
+    {
+        const auto params = eventPacket.getParameters();
+        const DataDescriptorPtr newSignalDescriptor = params[event_packet_param::DATA_DESCRIPTOR];
+        const DataDescriptorPtr newDomainDescriptor = params[event_packet_param::DOMAIN_DATA_DESCRIPTOR];
+
+        std::scoped_lock lock(signalMutex);
+
+        Bool changed = False;
+
+        if (newSignalDescriptor.assigned() && newSignalDescriptor != mirroredDataDescriptor)
+        {
+            mirroredDataDescriptor = newSignalDescriptor;
+            changed = True;
+
+            if (!this->coreEventMuted && this->coreEvent.assigned())
+            {
+                const auto args = createWithImplementation<ICoreEventArgs, CoreEventArgsImpl>(
+                    CoreEventId::DataDescriptorChanged, Dict<IString, IBaseObject>({{"DataDescriptor", mirroredDataDescriptor}}));
+
+                this->triggerCoreEvent(args);
+            }
```

Comment:
Is it okay to trigger this under a lock?

Reply:
Currently, it doesn't cause any issues, but I've anyway moved it outside the mutex lock to avoid potential problems in the future.